compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -1033,6 +1027,45 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         }
     }
 
+    fn assoc_ty_contraint_param_error_emit(
+        &self,
+        gen_args: &GenericArgs,
+        data: &ParenthesizedArgs,
+    ) -> () {
+        let mut err = self.sess.struct_span_err(
+            gen_args.span(),
+            "parenthesized generic arguments cannot be used in associated type constraints",
+        );
+        if let Ok(snippet) = self.sess.source_map().span_to_snippet(data.span)
```

Comment:
Please avoid re-parsing the source code, you have all the required information in the spans.
The span of the `(` is `data.span.shrink_to_lo().to(data.inputs.first().unwrap().span().shrink_to_lo())`.
Similarly computation for the span for `)`.
You can use a multipart suggestion to suggest replacing `(` by `<` and `)` by `>`, without touching to the args' code.